File: AMoAdiOSSdkDemo(Objective-C)/AMoAdiOSSdkDemo(Objective-C)/Regexp.m
//
//  Regexp.m
//  AMoAdiOSSdkDemo(Object-C)
//
//  Created by AMoAd on 2016/09/06.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

#import "Regexp.h"
@interface Regexp()
@property (nonatomic) NSString *pattern;
@end
@implementation Regexp

- (instancetype)init:(NSString *)pattern {
    if (self = [super init]) {
        self.pattern = pattern;
    }
    return self;
}

- (Boolean)isMatch:(NSString *)input {
    NSError* err = nil;
    NSRegularExpression* regex = [NSRegularExpression regularExpressionWithPattern:self.pattern options:NSRegularExpressionCaseInsensitive error:&err];
    NSTextCheckingResult *match = [regex firstMatchInString:input options:0 range:NSMakeRange(0, input.length)];
    if (match) {
        return true;
    }
    return false;
}
@end

File: AMoAdiOSSdkDemo/AMoAdiOSSdkDemo/Regexp.swift
//
//  Regexp.swift
//  AMoAdiOSSdkDemo
//
//  Created by AMoAd on 2016/08/15.
//  Copyright © 2016年 AMoAd. All rights reserved.
//

import Foundation

class Regexp {
    let internalRegexp: NSRegularExpression
    let pattern: String
    
    init(_ pattern: String) {
        self.pattern = pattern
        self.internalRegexp = try! NSRegularExpression( pattern: pattern, options: NSRegularExpression.Options.caseInsensitive)
    }
    
    func isMatch(_ input: String) -> Bool {
        let matches = self.internalRegexp.matches( in: input, options: [], range:NSMakeRange(0, input.characters.count) )
        return matches.count > 0
    }
    
    func matches(_ input: String) -> [String]? {
        if self.isMatch(input) {
            let matches = self.internalRegexp.matches( in: input, options: [], range:NSMakeRange(0, input.characters.count) )
            var results: [String] = []
            for i in 0 ..< matches.count {
                results.append( (input as NSString).substring(with: matches[i].range) )
            }
            return results
        }
        return nil
    }
}
